### DISCUSSION TITLE
How to add Recording Rules and Alerting Rules in mimir Ruler?

### DISCUSSION THREAD

**Discussion by luvpreetsingh** [grafana/mimir#2827] [Help and support] [answered]
Hi,

I am a bit confused about adding recording rules to mimir ruler? Does it use prometheus rules?

Can I add separate recordingrules and alertingrules to ruler using some configmap or a file? or maybe using some CRD?

How can I provision the rules without the UI?

**Comment by dimitarvdimitrov** [comment, marked as answer]
The ruler uses prometheus rules, yes. But the format for uploading them is very slightly different from the vanilla prometheus formal. You can use the `mimirtool rules load` or `mimirtool rules load` commands to create and delete rules. The [docs are here](https://grafana.com/docs/mimir/v2.1.x/operators-guide/tools/mimirtool/#rules)

**Reply by luvpreetsingh** [reply]
@dimitarvdimitrov I am thinking of using grafana dashboard for alerts.

Is it possible to add recording rules via grafana dashboard? Or is `mimirtool` the only way to add recording rules in mimir?

**Reply by dimitarvdimitrov** [reply]
It is possible to manage them vie the alerting plugin in grafana - https://grafana.com/docs/grafana/latest/alerting/alerting-rules/. But I believe this is what you refer to as "the UI". There are two open issues to add rules support to the chart - https://github.com/grafana/mimir/issues/2861 and https://github.com/grafana/helm-charts/issues/1240

**Reply by dminca** [reply]
I just landed here after trying to implement it in a GitOps fashion — now finding out this is not so _GitOps-friendly_ as expected initially 🤦

**Reply by Syndlex** [reply]
https://github.com/Syndlex/Openawareness-Operator @dminca  I took the time to develop a small operator that solves the Problems I had here. :)

**Reply by davimmt** [reply]
Hey, @Syndlex, what is the difference between your operator and Grafana Alloy's [mimir.rules.kubernetes](https://grafana.com/docs/alloy/latest/reference/components/mimir/mimir.rules.kubernetes/)/[mimir.alerts.kubernetes](https://grafana.com/docs/alloy/latest/reference/components/mimir/mimir.alerts.kubernetes/) components?

**Reply by Syndlex** [reply]
Ultimately nothing... 

I don't think that Alert,Dashboard,Rules Provisioning should be in a Telemetry collector. Architecturally we are abusing a Alloy here. 
Also under the hood I am using the Kubernetes Operator SDK.
